tests/modules.py

```diff
@@ -205,6 +206,71 @@ def test_npm_app(network, args):
         assert r.status_code == http.HTTPStatus.OK, r.status_code
         assert r.body["available"], r.body
 
+    return network
+
+
+@reqs.description("Test tsoa-based Node.js/npm app")
+def test_npm_tsoa_app(network, args):
+    primary, _ = network.find_nodes()
+
+    LOG.info("Building npm app")
+    app_dir = os.path.join(THIS_DIR, "npm-tsoa-app")
+    subprocess.run(["npm", "ci"], cwd=app_dir, check=True)
+    subprocess.run(["npm", "run", "build"], cwd=app_dir, check=True)
+
+    LOG.info("Deploying npm app modules")
+    module_name_prefix = "/my-tsoa-npm-app/"
+    dist_dir = os.path.join(app_dir, "dist")
+
+    proposal_body, _ = ccf.proposal_generator.update_modules(
+        module_name_prefix, dist_dir
+    )
+    proposal = network.consortium.get_any_active_member().propose(
+        primary, proposal_body
+    )
+    network.consortium.vote_using_majority(primary, proposal)
+
+    LOG.info("Deploying endpoint script")
+    metadata_path = os.path.join(dist_dir, "endpoints.json")
+    with open(metadata_path) as f:
+        metadata = json.load(f)
+    # Temporarily only until endpoints can be called directly without proxy.
+    app_script = "return {"
+    for url, methods in metadata["endpoints"].items():
+        for method, cfg in methods.items():
+            app_script += f"""
+            ["{method.upper()} {url[1:]}"] = [[
+                import {{ {cfg["js_function"]} as f }}
+                  from ".{module_name_prefix}{cfg["js_module"]}";
+                export default (request) => f(request);
+            ]],"""
+    app_script = app_script[:-1] + "\n}"
+
+    with tempfile.NamedTemporaryFile("w") as f:
+        f.write(app_script)
+        f.flush()
+        network.consortium.set_js_app(remote_node=primary, app_script_path=f.name)
+
+    LOG.info("Calling npm app endpoints")
+    with primary.client("user0") as c:
+        body = [1, 2, 3, 4]
+        r = c.post("/app/partition", body)
+        assert r.status_code == http.HTTPStatus.OK, r.status_code
+        assert r.body == [[1, 3], [2, 4]], r.body
+
+        r = c.post("/app/proto", body)
+        assert r.status_code == http.HTTPStatus.OK, r.status_code
+        assert r.headers["content-type"] == "application/x-protobuf"
+        # We could now decode the protobuf message but given all the machinery
```

Comment:
How is it encoded without this machinery?

Reply:
This PR uses the same endpoints as in the basic npm app. Any improvements etc to that should probably be a different issue/PR.

Reply:
I think you can use the reflection API to parser/extract the value the same way the npm app does it (ie. without code gen): https://googleapis.dev/python/protobuf/latest/google/protobuf/message.html#google.protobuf.message.Message

Reply:
The existing npm app doesn't do it either. I don't think it's essential to this test and it's not what people would do in reality.